SparseLinear.lua

```diff
@@ -51,6 +51,9 @@ function SparseLinear:reshapeInput(input)
 end
 
 function SparseLinear:updateOutput(input)
+   if self.sparseUpdate == ONE_LAST_INPUT then
+      self.sparseUpdate = ACC_MULTIPLE_TIMES
```

Comment:
Thanks for pointing this out, I'll patch it soon as I get home.

On Sun, Mar 20, 2016, 15:55 Soumith Chintala <notifications@github.com>
wrote:

> In SparseLinear.lua
> <https://github.com/torch/nn/pull/725#discussion_r56770668>:
>
> > @@ -51,6 +51,9 @@ function SparseLinear:reshapeInput(input)
> >  end
> >
> >  function SparseLinear:updateOutput(input)
> > +   if self.sparseUpdate == ONE_LAST_INPUT then
> > +      self.sparseUpdate = ACC_MULTIPLE_TIMES
>
> yes, if I understand, the intention is to only do sparse updates if you
> have a pattern of: forward + backward + update, forward + backward +
> update, ...
>
> However, after processing two mini-batches of FW + BW + UP, FW + BW + UP,
> the third mini-batch no longer sees sparse updates, as you never reset the
> state to NO_LAST_INPUT (because lines 195, 208)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/torch/nn/pull/725/files/1156255899c621e322a1f7991a817d4139af754d#r56770668>
>